File: src/dao/hibernate/Tb_userDao.java
package dao.hibernate;

import java.util.List;

import model.Tb_user;

import org.springframework.orm.hibernate3.support.HibernateDaoSupport;

import dao.interf.ITb_userDao;

public class Tb_userDao extends HibernateDaoSupport implements ITb_userDao
{
	// private HibernateTemplate ht;
	// private SessionFactory sessionFactory;
	//
	// private void setHibernateTemplate()
	// {
	// if (null == ht)
	// {
	// ht = new HibernateTemplate(this.sessionFactory);
	// }
	// }
	//
	// public void setSessionFactory(SessionFactory sessionFactory)
	// {
	// this.sessionFactory = sessionFactory;get
	// }

	public void delete(Tb_user model)
	{
		// setHibernateTemplate();
		// ht.delete(model);
		getHibernateTemplate().delete(model);
	}

	public void deleteById(int id)
	{
		// setHibernateTemplate();
		// ht.delete(model);
		Tb_user model = new Tb_user();
		model.setId(id);
		delete(model);
	}

	public Tb_user get(int id)
	{
		// Session session =
		// getHibernateTemplate().getSessionFactory().openSession();
		// session.get(Tb_user.class, new Integer(id));
		return (Tb_user) getHibernateTemplate().get(Tb_user.class, new Integer(id));
	}

	public void save(Tb_user model)
	{
		getHibernateTemplate().save(model);
	}

	public void update(Tb_user model)
	{
		getHibernateTemplate().update(model);
	}

	@SuppressWarnings("unchecked")
	public List getAll()
	{
		return getHibernateTemplate().find("from Tb_user");
	}
}

File: ash/ash/Dao/Hibernate/Tb_userDao.cs
using System;
using System.Collections;
using System.Linq;
using System.Web;
using NHibernate;
using ash.Dao.Interf;
using model;
using Spring.Transaction;
using Spring.Stereotype;
using Spring.Transaction.Interceptor;
using Spring.Data.NHibernate.Support;

namespace ash.Dao.Hibernate
{
    //[Repository]
    public class Tb_userDao : HibernateDaoSupport, ITb_userDao
    {
        //private HibernateTemplate ht;
        //public ISessionFactory sf { set; get; }

        //private void setHibernateTemplate()
        //{
        //    if (null == ht)
        //    {
        //        ht = new HibernateTemplate(this.sf);
        //        ht.TemplateFlushMode = TemplateFlushMode.Auto;
        //        ht.CacheQueries = true;
        //    }
        //}

        public void delete(User model)
        {
            //sf.GetCurrentSession().Delete(tb_user);
            HibernateTemplate.Delete(model);
        }

        public User get(int id)
        {
            //setHibernateTemplate();
            return HibernateTemplate.Get(typeof(User), id) as User;
        }

        //[Transaction(Spring.Transaction.TransactionPropagation.Required, ReadOnly = false)]
        //[Transaction]
        public void save(User model)
        {
            //setHibernateTemplate();
            //sf.GetCurrentSession().Save(person);
            //ht.Save(person);
            //ISession s = sf.OpenSession();
            //ITransaction t = s.BeginTransaction();
            //s.FlushMode = FlushMode.Auto;
            //s.Save(person);
            //t.Commit();
            //HibernateTemplate.TemplateFlushMode = TemplateFlushMode.Auto;
            HibernateTemplate.Save(model);

            //ISession s = HibernateTemplate.SessionFactory.OpenSession();
            //ITransaction t = s.BeginTransaction();
            //s.Save(model);
            //t.Commit();
            //s.Close();
        }

        public void update(User model)
        {
            //setHibernateTemplate();
            //HibernateTemplate.Update(person);
            HibernateTemplate.Update(model);
        }

        public IList getAll()
        {
            //setHibernateTemplate();
            //return HibernateTemplate.Find("from Tb_user");
            return HibernateTemplate.Find("from Tb_user");
        }
    }
}
